pkg/autoscaler/autoscaler.go

```diff
@@ -152,14 +152,21 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (int32, bool) {
 	// Use 1 if there are zero current pods.
 	readyPodsCount := math.Max(1, float64(originalReadyPodsCount))
 
-	observedStableConcurrency, observedPanicConcurrency, lastBucket := a.aggregateData(now, spec.MetricSpec.StableWindow, spec.MetricSpec.PanicWindow)
+	// Remove outdated data.
+	a.buckets.RemoveOlderThan(now.Add(-spec.MetricSpec.StableWindow))
 	if a.buckets.IsEmpty() {
 		logger.Debug("No data to scale on.")
 		return 0, false
 	}
 
-	// Log system totals.
-	logger.Debugf("Current concurrent clients: %0.3f", lastBucket.Sum())
+	// Compute data to scale on.
+	panicAverage := aggregation.Average{}
+	stableAverage := aggregation.Average{}
+	a.buckets.Process(now,
+		aggregation.Windowed(spec.MetricSpec.PanicWindow, panicAverage.Accumulate),
+		aggregation.Windowed(spec.MetricSpec.StableWindow, stableAverage.Accumulate))
```

Comment:
It feels that `Windowed` is redundant here, since just a few lines above you erased all the objects older than `StableWindow`?